Bug: #[url_patterns] and #[routes] url_prelude fail with multi-file view modules

## Description

When views use the per-file pattern (each endpoint in its own file under `views/`), two issues prevent `#[url_patterns]` and `#[routes]` url_prelude from working:

### Issue 1: `#[url_patterns]` generates incorrect `__url_resolver_*` import paths

`#[url_patterns]` generates `pub use super::views::__url_resolver_login::*;` when given `.endpoint(views::login)`. However, in the multi-file pattern:

```
views.rs          // pub mod login; pub use login::login;
views/login.rs    // #[post(...)] pub async fn login(...) { ... }
```

The `__url_resolver_login` module is generated inside `views/login.rs` (by the `#[post]` macro), so it lives at `views::login::__url_resolver_login`, NOT `views::__url_resolver_login`.

The macro needs to understand that `views::login` refers to a function re-exported from `views::login::login`, and look for `__url_resolver_login` in the `views::login` module instead.

### Issue 2: `#[routes]` generated url_prelude code misses `UrlResolver` trait import

The `__gen_<app>_method` macro generated by `#[routes]` calls `self.resolver.resolve_url(...)`, which requires the `UrlResolver` trait in scope. However, the generated code doesn't include a `use` statement for the trait.

Adding `use reinhardt::UrlResolver;` at the call site (config/urls.rs) doesn't help because the generated macro expands in its own scope.

**Fix**: The generated code should include `use reinhardt_urls::routers::resolver::UrlResolver;` inside each `impl` block or at the top of the generated `url_prelude` module.

### Reproduction

Project structure:
```
apps/auth/
├── urls.rs     // #[url_patterns("auth")] pub fn url_patterns() -> ServerRouter { ... .endpoint(views::login) }
├── views.rs    // pub mod login; pub use login::login;
└── views/
    └── login.rs  // #[post("/login/", name = "auth_login")] pub async fn login(...) { ... }
```

With `installed_apps!` configured and `#[routes]` (non-standalone) on `config/urls.rs`.

**Error 1** (from `#[url_patterns]`):
```
error[E0432]: unresolved import `super::views::__url_resolver_login`
```

**Error 2** (from `#[routes]` url_prelude, after workaround for Error 1):
```
error[E0599]: no method named `resolve_url` found for reference `&ResolvedUrls` in the current scope
  = note: this error originates in the macro `__gen_auth_method`
```

### Workaround

Use `#[routes(standalone)]` to skip url_prelude generation entirely. This loses the type-safe URL resolution but allows compilation.

### Environment

- Rust: nightly
- reinhardt-web: 87e1c078d (latest main)
- Project: reinhardt-cloud (dashboard crate)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-core/macros/src/routes_registration.rs
+++ b/crates/reinhardt-core/macros/src/routes_registration.rs
@@ -448,11 +448,14 @@ pub(crate) fn routes_impl(args: TokenStream, input: ItemFn) -> Result<TokenStrea
 					}
 
 					// Callback macro for __for_each_url_resolver to generate methods.
+					// Each arm imports UrlResolver trait to bring resolve_url() into
+					// scope. (Issue #3669)
 					macro_rules! #gen_method_macro {
 						// No params
 						($app_label:ident, $method:ident, $route_name:literal, ) => {
 							impl #urls_struct<'_> {
 								pub fn $method(&self) -> String {
+									use #reinhardt::UrlResolver as _;
 									self.resolver.resolve_url(
 										concat!(stringify!($app_label), ":", $route_name),
 										&[],
@@ -464,6 +467,7 @@ pub(crate) fn routes_impl(args: TokenStream, input: ItemFn) -> Result<TokenStrea
 						($app_label:ident, $method:ident, $route_name:literal, $p1:literal) => {
 							impl #urls_struct<'_> {
 								pub fn $method(&self, p1: &str) -> String {
+									use #reinhardt::UrlResolver as _;
 									self.resolver.resolve_url(
 										concat!(stringify!($app_label), ":", $route_name),
 										&[($p1, p1)],
@@ -475,6 +479,7 @@ pub(crate) fn routes_impl(args: TokenStream, input: ItemFn) -> Result<TokenStrea
 						($app_label:ident, $method:ident, $route_name:literal, $p1:literal, $p2:literal) => {
 							impl #urls_struct<'_> {
 								pub fn $method(&self, p1: &str, p2: &str) -> String {
+									use #reinhardt::UrlResolver as _;
 									self.resolver.resolve_url(
 										concat!(stringify!($app_label), ":", $route_name),
 										&[($p1, p1), ($p2, p2)],
@@ -486,6 +491,7 @@ pub(crate) fn routes_impl(args: TokenStream, input: ItemFn) -> Result<TokenStrea
 						($app_label:ident, $method:ident, $route_name:literal, $p1:literal, $p2:literal, $p3:literal) => {
 							impl #urls_struct<'_> {
 								pub fn $method(&self, p1: &str, p2: &str, p3: &str) -> String {
+									use #reinhardt::UrlResolver as _;
 									self.resolver.resolve_url(
 										concat!(stringify!($app_label), ":", $route_name),
 										&[($p1, p1), ($p2, p2), ($p3, p3)],
@@ -497,6 +503,7 @@ pub(crate) fn routes_impl(args: TokenStream, input: ItemFn) -> Result<TokenStrea
 						($app_label:ident, $method:ident, $route_name:literal, $p1:literal, $p2:literal, $p3:literal, $p4:literal) => {
 							impl #urls_struct<'_> {
 								pub fn $method(&self, p1: &str, p2: &str, p3: &str, p4: &str) -> String {
+									use #reinhardt::UrlResolver as _;
 									self.resolver.resolve_url(
 										concat!(stringify!($app_label), ":", $route_name),
 										&[($p1, p1), ($p2, p2), ($p3, p3), ($p4, p4)],
@@ -508,6 +515,7 @@ pub(crate) fn routes_impl(args: TokenStream, input: ItemFn) -> Result<TokenStrea
 						($app_label:ident, $method:ident, $route_name:literal, $p1:literal, $p2:literal, $p3:literal, $p4:literal, $p5:literal) => {
 							impl #urls_struct<'_> {
 								pub fn $method(&self, p1: &str, p2: &str, p3: &str, p4: &str, p5: &str) -> String {
+									use #reinhardt::UrlResolver as _;
 									self.resolver.resolve_url(
 										concat!(stringify!($app_label), ":", $route_name),
 										&[($p1, p1), ($p2, p2), ($p3, p3), ($p4, p4), ($p5, p5)],

